AZ3166/src/cores/arduino/az_iot/provisioning_client/deps/RIoT/Reference/RIoT/Core/RiotCore.c

```diff
@@ -88,8 +88,46 @@ int RiotStart(uint8_t *CDI, uint16_t CDILen, const char *RegistrationId)
     uint32_t            length;
     uint32_t            dcType;
 
-    // Update subject common of alias certificate TBD data to input registration Id
-    x509AliasTBSData.SubjectCommon = RegistrationId;
+    // Verify registrationId input and given an empty input, generate one based on MAC address and firmware version of your DevKit
+    int regIdLength = strlen(RegistrationId);
+    if (regIdLength == 0){
+        char * registrationId = (char*)malloc(AUTO_GEN_REGISTRATION_ID_MAX_LENGTH);
+        memset(registrationId, 0, AUTO_GEN_REGISTRATION_ID_MAX_LENGTH);
+        char * macAddress = GetBoardID();
+        char * fmVersion = getDevkitVersion();
+        LogInfo("DevKit Firmware Version: %s", fmVersion);
+        snprintf(registrationId, AUTO_GEN_REGISTRATION_ID_MAX_LENGTH, "%sv%s", macAddress, fmVersion);
+        
+        for(int i = 0; i < strlen(registrationId); i++){
+            if(registrationId[i] == '.'){
+                registrationId[i] = 'v';
+            }
+        }
+
+        // Update subject common of alias certificate TBD data to auto-generated registration Id
+        x509AliasTBSData.SubjectCommon = registrationId;
+    }
+    else if (regIdLength > REGISTRATION_ID_MAX_LENGTH){
+        LogError("Input value for registrationId in DPS.ino exceeds maximum.");
```

Comment:
also print REGISTRATION_ID_MAX_LENGTH here

Reply:
Actually maximum value has been mentioned in dps.ino. It is fine to add it here again...